remotes/docker/schema1/converter.go

```diff
@@ -351,6 +376,57 @@ func (c *Converter) fetchBlob(ctx context.Context, desc ocispec.Descriptor) erro
 	return nil
 }
 
+func (c *Converter) reuseLabelBlobState(ctx context.Context, desc ocispec.Descriptor) (bool, error) {
+	cinfo, err := c.contentStore.Info(ctx, desc.Digest)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get blob info")
+	}
+	desc.Size = cinfo.Size
+
+	lState, ok := cinfo.Labels[labelLegacySchema1BlobState]
```

Comment:
Re-use `"containerd.io/uncompressed"` here instead of using a structured value. Maybe we can have `containerd.io/legacy.schema1.empty` to determine if a tar layer is empty. In that past we have hardcoded the uncompressed empty tar digest but not sure if that is reliable enough.

For media type, could just use this logic... if uncompressed == blob `images.MediaTypeDockerSchema2Layer` else  `images.MediaTypeDockerSchema2LayerGzip`. Nothing else is supported for schema1, no need to store it

Reply:
Actually after reading back through this code, I don't believe empty is relevant at this part of the code.  The `empty` attribute is determined by manifest, which is always loaded first. If the blob was empty, fetch will never be attempted and this code would not be called. So this code will only be called on non-empty layers. For empty layers, they are only included as part of the history and not part of the hash id calculation for the layer, so they never have a relevant diffid.